src/Sulu/Component/Media/SystemCollections/SystemCollectionManager.php

```diff
@@ -0,0 +1,262 @@
+     * @return int
+     */
+    private function getUserId()
+    {
+        if (!$this->tokenProvider || ($token = $this->tokenProvider->getToken()) === null) {
+            return;
+        }
+
+        return $token->getUser()->getId();
+    }
+
+    /**
+     * Go thru configuration and build all system collections.
+     *
+     * @param string $locale
+     * @param int $userId
+     *
+     * @return array
+     */
+    private function buildSystemCollections($locale, $userId)
+    {
+        $root = $this->getOrCreateRoot('system_collections', 'System', $locale, $userId);
+
+        $collections = ['root' => $root->getId()];
+        foreach ($this->config as $namespaceKey => $namespaceItem) {
+            $namespace = $this->getOrCreateCollection(
+                $namespaceKey,
+                $namespaceItem['meta_title'],
+                $userId,
+                $root->getId()
+            );
+            $collections[$namespaceKey] = $namespace->getId();
+
+            foreach ($namespaceItem['collections'] as $collectionKey => $collectionItem) {
+                $key = sprintf('%s.%s', $namespaceKey, $collectionKey);
+                $collections[$key] = $this->getOrCreateCollection(
+                    $key,
+                    $collectionItem['meta_title'],
+                    $userId,
+                    $namespace->getId()
+                )->getId();
+            }
+        }
+
+        $this->entityManager->flush();
+
+        return $collections;
+    }
+
+    /**
+     * Finds or create a new system-collection namespace.
+     *
+     * @param string $namespace
+     * @param string $title
+     * @param string $locale
+     * @param int $userId
+     * @param int|null $parent id of parent collection or null for root.
+     *
+     * @return Collection
+     */
+    private function getOrCreateRoot($namespace, $title, $locale, $userId, $parent = null)
+    {
+        if (($collection = $this->collectionManager->getByKey($namespace, $locale)) !== null) {
+            $collection->setTitle($title);
+
+            return $collection;
+        }
+
+        return $this->createCollection($title, $namespace, $locale, $userId, $parent);
+    }
+
+    /**
+     * Finds or create a new system-collection.
+     *
+     * @param string $key
+     * @param array $localizedTitles
+     * @param int $userId
+     * @param int|null $parent id of parent collection or null for root.
+     *
+     * @return Collection
+     */
+    private function getOrCreateCollection($key, $localizedTitles, $userId, $parent)
+    {
+        $locales = array_keys($localizedTitles);
+        $firstLocale = array_shift($locales);
+
+        $collection = $this->collectionManager->getByKey($key, $firstLocale);
+        if ($collection === null) {
+            $collection = $this->createCollection($localizedTitles[$firstLocale], $key, $firstLocale, $userId, $parent);
+        } else {
+            $collection->setTitle($localizedTitles[$firstLocale]);
```

Comment:
but when we change the title in our bundles (or someone else) has to provide an upgrade script ... with this little line we have to do nothing ...

but i know something about ... i know that this collection is linked to this configuration with the key. an exception here is no solution because we need this to find the id of the systemcollection.